### PR TITLE
Implement LPOS

### LINKED ISSUE #253: LPOS is not supported command for LISTs
`LPOS` is the command to return the index of matching elements inside a Redis list. It doesn't seem to supported at the moment.
https://redis.io/commands/lpos

Wondering whether there are plans to add it.

### PR DESCRIPTION
**Pull request by ianstarz**
Fixes #253

### CODE REVIEW

**Comment by ianstarz** [comment]
I didn't know how to run the integration tests, but I manually tested integration using the redis CLI on the redis docs. Will integration tests be run through CI?

**Comment by alicebob** [comment]
Hi! Thanks for the PR! It looks good after a quick look, but I'll have a proper look soon.

CI I need to approve manually says github, which I did.

File: cmd_list.go [hunk 1]
```diff
@@ -165,6 +166,148 @@ func (m *Miniredis) cmdLindex(c *server.Peer, cmd string, args []string) {
...
+		var countSpecified bool
+		if len(args) > 2 {
+			for i := 2; i < len(args); i++ {
+				if i%2 == 0 {
+					val := args[i+1]
+					var err error
+					switch strings.ToLower(args[i]) {
+					case "rank":
+						if rank, err = strconv.Atoi(val); err != nil {
+							c.WriteError(msgInvalidInt)
```
Comment (alicebob) [inline cmd_list.go:221]: All these `c.WriteError()`s need to call `setDirty(c)` (for transactions) before returning.

**Review by alicebob — CHANGES_REQUESTED** [review]
Looks good, only the comment I made (with edit!).

File: cmd_list.go (hunk 1 above)
Comment (ianstarz) [reply cmd_list.go]: Done.

**Comment by ianstarz** [comment]
Thank you for the quick review! Comment resolved; please take another look.

**Comment by alicebob** [comment]
Looks good, I'll have a 2nd (awake) look again tomo. Can you squash the last three commits already? That way you get the credits. (or squash all 4 if that's easier, doesn't matter much)

**Comment by ianstarz** [comment]
Squashed.

**Comment by alicebob** [comment]
Thanks! I'll make it a proper release when I've got the other open PRs in.

**Comment by ianstarz** [comment]
Thank you for the release!

**Pull request merged**